cpp/include/cuspatial/cubic_spline.hpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <cudf/column/column.hpp>
+#include <cudf/table/table.hpp>
+
+namespace cuspatial {
+
+/**
+ * @brief Compute quadratic interpolations of a set of points based on their
+ * ids and a coefficient matrix.
+ *
+ * @param[in] query_points float32 column of coordinate values to be 
```

Comment:
Why are these called "query points"? Aren't they just vertices of the splines? Is there a query happening? If not, suggest renaming. Are these coordinates 1D? 2D? 3D? If multidimensional, are x and y stored together? Specify.

```suggestion
 * @param[in] coordinates column of coordinate values to be 
```

Reply:
They are query points because they are computed through two steps - a (linear) search to find which set of four coefficients to use in the interpolation, and then the interpolation.